### PR TITLE
chore(deps): Use MUI 5

### PR DESCRIPTION
**Pull request by jeremyckahn**
### What this PR does

This PR upgrades Farmhand from MUI 4 to MUI 5. Necessary code changes are made, as well as a handful of UI updates to support the MUI 5 component changes.

### How this change can be validated

This is a sweeping change, so a wide-ranging smoke test would be best.

### Questions or concerns about this change

I've tried to thoroughly test this and fix regressions, but it's possible I missed something. For a change this foundational, I think we need to accept that we will likely introduce minor regressions. We can fix them as we find them. Hopefully I didn't miss anything significant!

### Additional information

I recommend hiding whitespace changes when reviewing this.

I followed the MUI migration guide for this upgrade: https://mui.com/material-ui/migration/migration-v4/

The majority of this PR is [updating imports](https://mui.com/material-ui/migration/migration-v4/#replace-all-imports).

The most significant UI update is the design of the tabs. This is the old tabs component:

Old tabs screenshot

And this is the new one:

New tabs screenshot

### CODE REVIEW

**Comment by lstebner** [comment]
image
 these tooltips are behaving a bit differently now. when they show up and i move the mouse to the next tool, the tooltip seems to get focus and i have to mouse all the way off, then back, to click a tool the tip is blocking. maybe we can move them to the left more so they never block the tools? on prod right now this doesn't seem to happen

**Review by lstebner — APPROVED** [review]
changes all seem straightforward and the tooltip thing i posted is the only issue i'm seeing with the changes. given that mobile is probably a bigger market than desktop i don't think it's a blocker but would be nice to improve down the road. onwards to styled components!

File: src/components/Farmhand/Farmhand.js [hunk 1]
```diff
@@ -1315,131 +1315,136 @@ export default class Farmhand extends FarmhandReducers {
...
-                        'aria-label': 'Next view',
-                        color: 'primary',
-                        onClick: () => this.focusNextView(),
+                        'aria-label':
+                          'End the day to save your progress and advance the game.',
+                        className: 'end-day',
+                        color: 'error',
+                        onClick: handlers.handleClickEndDayButton,
+                        sx: {
+                          zIndex: 1100,
```
[the remark this answers is not in the document]
Comment (jeremyckahn) [reply src/components/Farmhand/Farmhand.js]: Good idea. Done in https://github.com/jeremyckahn/farmhand/commit/8241411f12ab69807de6312c2af24ca01de0e2ee!

**Review by jeremyckahn — COMMENTED** [review]
> image 
> these tooltips are behaving a bit differently now. when they show up and i move the mouse to the next tool, the tooltip seems to get focus and i have to mouse all the way off, then back, to click a tool the tip is blocking. maybe we can move them to the left more so they never block the tools? on prod right now this doesn't seem to happen

Great catch! https://github.com/jeremyckahn/farmhand/commit/4bb3da2e90fa17e592ec552835326bbc3ee81f5d changes Field tooltips to follow the cursor, which mitigates this issue and works well in all screen orientations (such as portrait, where the tool selection UI is at the bottom of the screen).

**Review by lstebner — APPROVED** [review]
thanks for making those updates @jeremyckahn ! the tooltip behavior is _much_ nicer now. i'm glad there was an easy fix for that one. let's ship this!

**Pull request merged**